mantidimaging/gui/windows/live_viewer/model.py

```diff
@@ -6,15 +6,79 @@
 from typing import TYPE_CHECKING
 from pathlib import Path
 from logging import getLogger
+
+import numpy as np
 from PyQt5.QtCore import QFileSystemWatcher, QObject, pyqtSignal, QTimer
 
+from tifffile import tifffile
+from astropy.io import fits
+
+from mantidimaging.core.utility.sensible_roi import SensibleROI
+
 if TYPE_CHECKING:
     from os import stat_result
     from mantidimaging.gui.windows.live_viewer.view import LiveViewerWindowPresenter
 
 LOG = getLogger(__name__)
 
 
+def load_image_from_path(image_path: Path) -> np.ndarray:
+    """
+    Load a .Tif, .Tiff or .Fits file only if it exists
+    and returns as an ndarray
+    """
+    if image_path.suffix.lower() in [".tif", ".tiff"]:
+        with tifffile.TiffFile(image_path) as tif:
+            image_data = tif.asarray()
+    elif image_path.suffix.lower() == ".fits":
+        with fits.open(image_path.__str__()) as fit:
+            image_data = fit[0].data
+    return image_data
+
+
+class ImageCache:
+    """
+    An ImageCache class to be used as a decorator on image read functions to store recent images in memory
+    """
+    cache_dict: dict[Image_Data, tuple[np.ndarray, float]]
+    max_cache_size: int | None = None
+
+    def __init__(self, max_cache_size=None):
+        self.max_cache_size = max_cache_size
+        self.cache_dict = {}
+
+    def add_to_cache(self, image: Image_Data, image_array: np.ndarray):
+        if image not in self.cache_dict.keys():
+            if self.max_cache_size is not None:
+                if self.max_cache_size <= len(self.cache_dict):
+                    self.remove_oldest_image()
+            self.cache_dict[image] = (image_array, image.image_modified_time)
+
+    def remove_from_cache(self, image: Image_Data):
+        if image in self.cache_dict:
+            del self.cache_dict[image]
+
+    def get_oldest_image(self):
+        time_ordered_cache = sorted(self.cache_dict.items(), key=lambda item: item[1][-1])
```

Comment:
You can used the modification time from the dictionary key to simplify this
```
time_ordered_cache = sorted(self.cache_dict.keys(), key=lambda item: item.image_modified_time)
return time_ordered_cache[0]
```

Then the dictionary does not need the mod time to be put in the value and can be `cache_dict: dict[Image_Data, np.ndarray]`.

You can also use `key=attrgetter('image_modified_time')`, see https://docs.python.org/3/howto/sorting.html#operator-module-functions-and-partial-function-evaluation

Also you can use `min` instead of `sorted`, which is faster

`return min(self.cache_dict.keys(), key=attrgetter('image_modified_time'))`